File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/library/items/media/files/uri/IFileUriProvider.java
package com.lasthopesoftware.bluewater.client.library.items.media.files.uri;

import android.net.Uri;
import com.lasthopesoftware.bluewater.client.library.items.media.files.ServiceFile;
import com.namehillsoftware.handoff.promises.Promise;

public interface IFileUriProvider {
	Promise<Uri> promiseFileUri(ServiceFile serviceFile);
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/browsing/items/media/files/uri/IFileUriProvider.kt
package com.lasthopesoftware.bluewater.client.browsing.items.media.files.uri

import android.net.Uri
import com.lasthopesoftware.bluewater.client.browsing.items.media.files.ServiceFile
import com.namehillsoftware.handoff.promises.Promise

fun interface IFileUriProvider {
    fun promiseFileUri(serviceFile: ServiceFile): Promise<Uri?>
}
